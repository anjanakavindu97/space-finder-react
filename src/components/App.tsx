import React from 'react';
import { User } from '../models/Model'
import { AuthService } from '../services/AuthService'
import { Loging } from './Login';

interface AppState{
  user: User | undefined
}

export class App extends React.Component<{}, AppState>{

  private authService: AuthService = new AuthService();

  constructor(props: any) {
    super(props)

    this.setUser = this.setUser.bind(this)
  }

  private setUser(user: User) {
    this.setState({
      user: user
    })
    console.log('setting the user ' + user)
  }

  render() {
    return (
      <div>
        App from class work....
        <Loging authService={this.authService} setUser={this.setUser}/>
      </div>
    );
  }
}
