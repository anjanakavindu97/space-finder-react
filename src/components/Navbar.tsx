import React from "react";
import { User } from "../models/Model";
import { Link } from "react-router-dom";

export class Navbar extends React.Component<{user: User | undefined}> {

    render() {
        let loginLogout: any;
        if (this.props.user) {
            loginLogout = <Link to='/logout' style={{float:"right"}}>{this.props.user.userName}</Link>
        } else {
            loginLogout = <Link data-testid='login-link' to='/login' style={{float:"right"}}>Login</Link>
        }

        return (
            <div className="navbar">
                <Link data-testid='home-link' to='/'> Home</Link>
                <Link data-testid='profile-link' to='/profile'> Profile</Link>
                <Link data-testid='space-link' to='/spaces'> Spaces</Link>
                { loginLogout }
            </div>
        )
    }
}