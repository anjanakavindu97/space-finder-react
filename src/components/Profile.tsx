import React from "react";
import { User, UserAttribute } from "../models/Model";
import { AuthService } from "../services/AuthService";
import { Link } from "react-router-dom";

interface ProfileState {
    userAttributes: UserAttribute[]
}

interface ProfileProps {
    user: User | undefined
    authService: AuthService
}

export class Profile extends React.Component<ProfileProps, ProfileState> {

    state: ProfileState = {
        userAttributes: []
    }

    async componentDidMount() {
        if (this.props.user) {
            const userAtrs = await this.props.authService.getUserAttributes(this.props.user)
            this.setState({
                userAttributes: userAtrs
            })
        }
    }

    private renderUserAttributes() {
        const rows = []
        for (const userAttribute of this.state.userAttributes) {
            rows.push(<tr key={userAttribute.Name.toString()}>
                <td>{userAttribute.Name}</td>
                <td>{userAttribute.Value}</td>
            </tr>)
        }
        return <table>
            <tbody>
                {rows}
            </tbody>
        </table>
    }
    render() {
        let ProfileSpace
        if (this.props.user) {
            ProfileSpace = <div>
                <h3>Hello {this.props.user.userName}</h3>
                Here are your attributes
                {this.renderUserAttributes()}
                </div>
        } else {
            ProfileSpace = <div>
                Please <Link to='/login'>Login</Link>
                </div>
        }

        return (
            <div>
                Welcome to your profile
                {ProfileSpace}
            </div>
        )
    }
}