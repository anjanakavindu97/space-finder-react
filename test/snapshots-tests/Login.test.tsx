import React from 'react';
import { Login } from '../../src/components/Login';
import ReactTestRenderer from 'react-test-renderer';

describe('Login component snapshot test', () => {

    test('initial test', () => {
        const snap = ReactTestRenderer.create(
        <Login authService={{} as any} setUser={{} as any} />);

        expect(snap).toMatchSnapshot();
    });
});