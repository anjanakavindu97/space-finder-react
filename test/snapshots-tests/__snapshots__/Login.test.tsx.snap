// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component snapshot test initial test 1`] = `
<div>
  <h2>
    Please Loging
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value=""
    />
    <br />
    <input
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
